net/tcp/tcp_send.c

```diff
@@ -372,6 +384,13 @@ static void tcp_sendcommon(FAR struct net_driver_s *dev,
 
       /* Update the Receiver Window */
 
+#ifdef CONFIG_NET_TCP_WINDOW_SCALE
+      if (recvwndo > 0 && conn->rcv_scale > 0)
+        {
+          recvwndo <<= conn->rcv_scale;
+        }
+#endif
+
```

Comment:
adjusting the window for UINT16_MAX and scaling here can confuse tcp_should_send_recvwindow.
i feel it's simpler to do it in tcp_get_recvwindow.
how do you think?